lib/include/jxl/encode.h

```diff
@@ -234,6 +234,17 @@ JXL_EXPORT JxlEncoderStatus JxlEncoderSetICCProfile(JxlEncoder* enc,
                                                     const uint8_t* icc_profile,
                                                     size_t size);
 
+/**
+ * Initializes a JxlBasicInfo struct to default values.
+ * For forwards-compatibility, this function has to be called before values
```

Comment:
The problem with `= {}` is that it initializes everything to zeroes, which does not make sense for various fields (bitdepth, orientation, num_chans ...). Requiring people to explicitly initialize width and height is one thing, requiring them to explicitly initialize all possible fields including those that have common defaults (8-bit non-oriented RGB, for example) is another thing. Especially considering for example that at the moment we ignore some of the fields in the encoder api (max nits, for example), so it is now fine if they are zero-initialized, but things will break if we actually start not ignoring those fields.

The struct is not opaque but it's also not quite complete (we haven't exposed all bitstream fields yet, and also there might be future header extensions resulting in new fields), so it's likely that we'll want to add more fields in the future. That's why padding is probably a good idea here, and also an explicit init function so that if in the future we add fields that need to get a sensible default when encoding, we can do that without breaking stuff.

Reply:
I also agree that having a Init function required to call is the correct way to go in C, and it was an oversight that the API did not have it before